src/librustc_middle/ty/sty.rs

```diff
@@ -864,6 +863,19 @@ impl<T> Binder<T> {
         Binder(value)
     }
 
+    /// Wraps `value` in a binder without actually binding any currently
+    /// unbound variables.
```

Comment:
We should add a note like this: This will shift any the debruijn indices of any escaping bound variables by 1 to avoid accidental capture.